### DISCUSSION TITLE
I am confused.

### DISCUSSION THREAD

**Discussion by xmyoard** [GriefPrevention/GriefPrevention#1557] [General]
Grief Prevention plugin is a very useful plugin but its too hard to understand its mechanics and config,

In my server players get 1000 more and more claimable blocks!!! after some time.

then I saw some lines in the config:

Claim Blocks Accrued Per Hour:
      Default: 1000

What I need is, All players shd have a max of 5000 blocks not more than that! 
How to do that?

**Comment by bigpresh** [comment]
Well, that config setting controls how blocks are given over time, so that players who've played for longer get to claim more land, and newbies who've just joined get less, which is a sensible mechanic so that people can't just join, claim loads of stuff then leave, but players who've been with you for some time and want to build lots, can.

However, if you don't want that, just change the `1000` there to a `0` and they won't be given any more blocks over time, then set `Claims.InitialBlocks: 5000` and you're done.  Or, if you want to keep the accrual over time, but limit the total they can reach, then just set `Claims.MaxAccruedBlocks: 5000` instead.

This is all well documented - see [claim limits in Setup and Configuration](https://github.com/TechFortress/GriefPrevention/wiki/Setup-and-Configuration#claim-limits) on the wiki.

5000 is a pretty miserly limit IMO, unless you're on some artifically limited map size for some reason.

**Reply by xmyoard** [reply]
hmph, I have done something.

Now My Config looks like this:

InitialBlocks: 5000
    Claim Blocks Accrued Per Hour:
      Default: 0
    Max Accrued Claim Blocks:
      Default: 5000

Now it's ok, right?

**Comment by bigpresh** [comment]
Not quite... you need to use the actual names as documented at the link I sent, i.e. for instance `BlocksAccruedPerHour` not `Claim Blocks Accrued Per Hour`, `MaxAccruedBlocks` not `Max Accrued Claim Blocks`. Also, you're missing correct indentation, although that may just be because you didn't wrap it in code markers here on GitHub so it's not rendering correctly.

I'd expect the whole section to look something like:

```
Claims:
  InitialBlocks: 5000
  BlocksAccruedPerHour:
    Default: 0
  MaxAccruedBlocks:
    Default: 5000
```

EDIT: except the actual config file on one of my servers does say `Max Accrued Claim Blocks`, not `MaxAccruedBlocks` so our doco may be wrong there... need to look at that later when I'm not working.

**Reply by xmyoard** [reply]
This makes me more confused.

Edit: Just say what should I do actually?

**Reply by bigpresh** [reply]
OK, having had a moment to look at the parsing code, the discrepancy between `MaxAccruedBlocks` and `Max Accrued Claim Blocks` is because the former is the name of the default setting, a straight int, and the latter is the hashref one that lets you set a value for `Default` and potentially others by permission groups etc, which does not seem to be documented.  The parsing code does this:

```java
this.config_claims_blocksAccruedPerHour_default = config.getInt("GriefPrevention.Claims.BlocksAccruedPerHour", 100);
this.config_claims_blocksAccruedPerHour_default = config.getInt("GriefPrevention.Claims.Claim Blocks Accrued Per Hour.Default", config_claims_blocksAccruedPerHour_default)
this.config_claims_maxAccruedBlocks_default = config.getInt("GriefPrevention.Claims.MaxAccruedBlocks", 80000);
this.config_claims_maxAccruedBlocks_default = config.getInt("GriefPrevention.Claims.Max Accrued Claim Blocks.Default", this.config_claims_maxAccruedBlocks_default);
```
So, you need to be setting `Claims.MaxAccruedBlocks`, or `Claims.Max Accrued Claim Blocks.Default`. 

So, on to what you should do - firstly, the documentation does explain the meaning of each setting.

However, you probably want to do this:
* find the `InitialBlocks` setting in your config and set it to 5000 so players start with 5000 blocks
* if you find the `MaxAccruedBlocks` setting, set it to 5000; if not, find the `Max Accrued Claim Blocks` setting, under which you'll see `Default` and a value, and change that value to 5000.
* You *can* also set `BlocksAccruedPerHour` to 0, but don't really need to, as it won't give them any blocks as they already have the maximum allowed.

If I get a moment, I'll tweak the docs on the wiki to explain the difference between `MaxAccruedBlocks` and `Max Accrued Claim Blocks` and 'BlocksAccruedPerHour` and `Claim Blocks Accrued Per Hour` as it's not very useful at the moment - it would be nice if the names we write out into the default config match the documentation!

**Reply by xmyoard** [reply]
This makes sense so let me give it a look.

thanks for the quick support!!
